service/Program.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Azure.IoT.ModelsRepository;
+using Microsoft.Azure.Devices;
+using Microsoft.Azure.Devices.Serialization;
+using Microsoft.Azure.DigitalTwins.Parser;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace service
+{
+
+    class Program
+    {
+        static readonly string cs = Environment.GetEnvironmentVariable("HUB_CS");
+        static readonly DigitalTwinClient dtc = DigitalTwinClient.CreateFromConnectionString(cs);
+        static readonly string deviceId = "self2";
+
+        static async Task Main(string[] args)
+        {
+            var twinResponse = await dtc.GetDigitalTwinAsync<BasicDigitalTwin>(deviceId);
+            Console.WriteLine($"Device '{deviceId}' announced: {twinResponse.Body.Metadata.ModelId}\n");
+
+            var model = await ResolveAndParse(twinResponse.Body);
+            Console.WriteLine("\nModel Parsed !! \n\n");
+            model.ToList().ForEach(i => Console.WriteLine(i.Key));
+
+            Console.ReadLine();
+        }
+
+        private static async Task<IReadOnlyDictionary<Dtmi, DTEntityInfo>> ResolveAndParse(BasicDigitalTwin twin)
+        {
+            IReadOnlyDictionary<Dtmi, DTEntityInfo> model;
+
+            string repo = "https://raw.githubusercontent.com/iotmodels/iot-plugandplay-models/selfdescribing";
```

Comment:
Consider moving initialization of the ModelParser to the class level (cached), or if you want to keep it JIT, then to the if condition that uses it.

Initializing it here is confusing because I'm expecting to be looking at how to do it for self-describing devices, in which case the model repo is irrelevant.

Reply:
is not irrelevant, since the targetModel extends the std interface that must be resolved from the public repo.

The instance is used in the two branches of the if condition, I dont want to move it to the class level as the goal of the `ResolveAndParse` method.

Note that I use the parsed model to check for extends

Reply:
Got it.

This is another great opportunity for code comments.